wagtailio/areweheadlessyet/thread_pool.py

```diff
@@ -33,9 +37,17 @@ def deploy(sender, **kwargs):
     """Triggers a build on Vercel."""
 
     try:
-        requests.post(
+        logger.info("Triggering build on Vercel.")
+        response = requests.post(
             settings.VERCEL_DEPLOY_HOOK_URL,
             timeout=settings.VERCEL_DEPLOY_REQUEST_TIMEOUT,
         )
+        response.raise_for_status()
+        logger.info("Build triggered on Vercel.")
+
     except requests.exceptions.Timeout:
-        pass  # Ignore this error
+        logger.warning("The request to trigger a new build on Vercel has timed out.")
+        return  # Ignore this error
+
+    except Exception:
```

Comment:
Nitpick: Whenever writing ``except`` clauses that catch all exceptions, I'd recommend always adding a separate ``except`` clause to make sure ``KeyboardInterrupt`` and ``SystemExit`` don't get caught.

So just above this line, add:

```python
except (KeyboardInterrupt, SystemExit):
    raise
```

It's unlikely to be needed here, but I've found this to be a good practise to always add as sometimes it can be needed when you don't expect it!

Reply:
Thanks!